src/entity/api_token.rs

```diff
@@ -109,14 +112,37 @@ impl Related<Memberships> for Entity {
 
 impl ActiveModelBehavior for ActiveModel {}
 
+fn encode_token(id: Uuid, token: &[u8; 32]) -> String {
+    format!(
+        "{TOKEN_IDENTIFIER}{}",
+        URL_SAFE_NO_PAD.encode(
+            id.as_bytes()
+                .iter()
+                .chain(token.iter())
+                .copied()
+                .collect::<Vec<u8>>(),
+        )
+    )
+}
+
+fn decode_token(token: &str) -> Option<(Uuid, [u8; 32])> {
```

Comment:
optional: could we create a newtype for API tokens? It would likely make our APIs easier to follow, but I'm not sure how much would be gained overall.

Reply:
I don't think that would improve the code